### PR TITLE
BUG: do not force emulation of 128-bit arithmetic.

### LINKED ISSUE #13857: Do not force 128-bit emulation for PCG64
I noticed that we are [forcing 128-bit arithmetic emulation](https://github.com/numpy/numpy/blob/master/numpy/random/setup.py#L61) even when it could be supported. I assume this was a temporary measure, but `pcg64.h` appears to not be written correctly without emulation for some reason. @bashtage @mattip Can you clarify?

### PR DESCRIPTION
**Pull request by rkern**
Fixes #13857 

I removed the check in the `setup.py` file as I think the logic in the header file does sufficient checks already. We'll see what the CI says.

### CODE REVIEW

**Comment by rkern** [comment]
@bashtage The performance tables will need to be rebuilt. I wondered why the `PCG64` performance was so mediocre this last update.

**Comment by bashtage** [comment]
@charris This is ready.  I'll regen the relevant columns in the table when merged.

File: numpy/random/src/pcg64/pcg64.h [hunk 1]
```diff
@@ -207,6 +209,13 @@ static inline uint64_t pcg_output_xsl_rr_128_64(pcg128_t state) {
                      state >> 122u);
 }
 
+static inline uint64_t
```
Comment (bashtage) [inline numpy/random/src/pcg64/pcg64.h]: Not a bug deal, but should these files be NumPy-ified? e.g. inline->NPY_INLINE ?  I think everything supports plain old inline now, so not strictly necessary.

Comment (charris) [reply numpy/random/src/pcg64/pcg64.h]: Yes, now that we require C99, inline should be fine even though it violates my sense of consistency :) At some point we should probably replace all the `NPY_INLINE` macros with `inline` just to make it a bit easier for new contributors who may know C but not NPY.

**Comment by charris** [comment]
Thanks Robert.

**Pull request merged**